test/alternative_dialog_spec.rb

```diff
@@ -0,0 +1,91 @@
+require_relative "spec_helper.rb"
+require "update-alternatives/UI/alternative_dialog"
+require "update-alternatives/model/alternative"
+
+describe UpdateAlternatives::AlternativeDialog do
+  def mock_ui_events(*events)
+    allow(Yast::UI).to receive(:UserInput).and_return(*events)
+  end
+
+  def simulate_the_change_of_selected_choice
+    allow(Yast::UI).to receive(:QueryWidget).with(:choices_table, :CurrentItem)
+      .and_return(alternative.value, "/usr/bin/vim")
+  end
```

Comment:
Ah, now I got it! You check it twice. One when opening the dialog (and you want it to be `alternative.value`) and then again when you are pretending you changed it. As said, not very intuitive. :smile: